### ISSUE TITLE
FR: add toleration for `node.kubernetes.io/network-unavailable` to install job

### ISSUE THREAD

**Issue by nicolas2bonfils** [k3s-io/helm-controller#247]
# Introduction

When a user choose not to enable flannel backend, the node is taint with `node.kubernetes.io/network-unavailable`.
Even with bootstrap flag, the install job does not have that [toleration](https://github.com/k3s-io/helm-controller/blob/333b5afe1beb8d555e7ccd6857fc268cec2b0b89/pkg/controllers/chart/chart.go#L584). Therefore, it is not possible to install a CNI (cilium for example) with k3s helm-controller.

# Proposed Changes
## 1. New toleration

Add a new toleration `node.kubernetes.io/network-unavailable` to the job YAML.

## 2. Implementation

```go
job.Spec.Template.Spec.Tolerations = []corev1.Toleration{
[...]
      {
	Key:    corev1.TaintNodeNetworkUnavailable,
	Effect: corev1.TaintEffectNoSchedule,
      },
[...]
```

NB : I'm working on a PR

**Comment by brandond** [comment]
I don't believe this is necessary. We use bootstrap charts to install CNIs using helm-controller for every CNI bundled with RKE2. Obviously this wouldn't be possible if bootstrap charts didn't work before the CNI was up.

**Comment by nicolas2bonfils** [comment]
I did more debug and find the kind of a race condition.

### context
a cluster without cloud controller and without network 
```
  --flannel-backend=none \
  --disable-kube-proxy \
  --disable-network-policy \
  --disable-cloud-controller \
  --kube-controller-manager-arg="--cloud-provider=external" \
  --kubelet-arg="--cloud-provider=external"
```

### steps
1. install k3s
2. install external cloud controller (GCP in my case)
3. install CNI (cilium helm chart with k3s helm controller)

### results
cluster starts with taints : `node.kubernetes.io/not-ready` and `node.cloudprovider.kubernetes.io/uninitialized`.
When CCM install finished, the cloud taint is removed. If there is no CNI at that moment, taint `node.kubernetes.io/network-unavailable` is added.
It can happen due to cni misconfiguration or execution delays.


If the job supports that state, it will allow the problem to resolve.

**Comment by brandond** [comment]
Hmm, interesting. I've not seen that one before, but I haven't tested much with the GCP cloud controller manager.

A PR would be welcome, if you're up for opening one. Should be easy enough to add here:
https://github.com/k3s-io/helm-controller/blob/333b5afe1beb8d555e7ccd6857fc268cec2b0b89/pkg/controllers/chart/chart.go#L584-L609

You can reference it as `{Key: TaintNodeNetworkUnavailable, Effect: corev1.TaintEffectNoSchedule}`

**Comment by nicolas2bonfils** [comment]
I close because the toleration was added in k3s-io/k3s#10831.
Thanks @brandond
